onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -302,12 +302,12 @@ inline void RegisterExecutionProvider(InferenceSession* sess, onnxruntime::IExec
   OrtPybindThrowIfError(sess->RegisterExecutionProvider(std::move(p)));
 }
 
-// ordered by default priority. highest to lowest.
+// ordered by default priority from highest to lowest. kCpuExecutionProvider should always be last.
 const std::vector<std::string>& GetAllProviders() {
-  static std::vector<std::string> all_providers = {kTensorrtExecutionProvider, kCudaExecutionProvider, kDnnlExecutionProvider,
-                                                   kNGraphExecutionProvider, kOpenVINOExecutionProvider, kNupharExecutionProvider,
-                                                   kVitisAIExecutionProvider, kCpuExecutionProvider, kMIGraphXExecutionProvider,
-                                                   kAclExecutionProvider, kArmNNExecutionProvider};
+  static std::vector<std::string> all_providers = {kTensorrtExecutionProvider, kCudaExecutionProvider, kMIGraphXExecutionProvider,
```

Comment:
Do you think it makes sense to have all providers defined in constants.h, so that it can be used by other language APIs, in case we need GetAllProviders() in those languages?

Reply:
only python has the concept of default EP registrations. the python api for EP registrations is kind of different than the others. So I don't think we should move it there yet. Let's revisit later.